.github/workflows/main.yml

```diff
@@ -38,10 +25,10 @@ jobs:
           architecture: x64
       - name: Install python packages
         run: |
-          pip install joblib==${{ matrix.JOBLIB_VERSION }} scikit-learn>=0.23.1 pytest pylint pyspark==${{ matrix.PYSPARK_VERSION }}
+          pip install joblib==${{ matrix.JOBLIB_VERSION }} scikit-learn>=0.23.1 pytest pylint "pyspark[connect]==${{ matrix.PYSPARK_VERSION }}" pandas
       - name: Run pylint
         run: |
           ./run-pylint.sh
       - name: Run test suites
         run: |
-          PYSPARK_PIN_THREAD=${{ matrix.PIN_MODE }} ./run-tests.sh
+          SPARK_CONNECT_MODE=${{ matrix.SPARK_CONNECT_MODE }} PYSPARK_VERSION=${{ matrix.PYSPARK_VERSION }} ./run-tests.sh
```

Comment:
The name `SPARK_CONNECT_MODE` conflicts with the one used in PySpark itself. I think we should better use a different name if you intend to use it for a diff reason.